pkg/backend/token_authcode.go

```diff
@@ -41,29 +38,41 @@ type refreshDescriptor struct {
 var _ scheduler.Descriptor = &refreshDescriptor{}
 
 func (rd *refreshDescriptor) Run(ctx context.Context, pc chan<- scheduler.Process) error {
-	ticker := rd.backend.clock.NewTicker(refreshInterval)
-	defer ticker.Stop()
+	// start out checking once per minute
+	refreshInterval := 60
+	ticker := rd.backend.clock.NewTicker(time.Duration(refreshInterval) * time.Second)
 
 	for {
-		err := rd.backend.data.Managers(rd.storage).AuthCode().ForEachAuthCodeKey(ctx, func(keyer persistence.AuthCodeKeyer) {
-			proc := &refreshProcess{
-				backend: rd.backend,
-				storage: rd.storage,
-				keyer:   keyer,
+		c, err := rd.backend.getCache(ctx, rd.storage)
+		if err == nil && c != nil && c.Config.RefreshInterval != 0 {
+			if c.Config.RefreshInterval != refreshInterval {
+				refreshInterval = c.Config.RefreshInterval
+				ticker.Stop()
+				ticker = rd.backend.clock.NewTicker(time.Duration(refreshInterval) * time.Second)
```

Comment:
As a note to myself: It would be nice to use Go 1.15's new `ticker.Reset()` method here but we don't support it yet in Leg or the Kubernetes API Machinery package we use for testing, sadly.